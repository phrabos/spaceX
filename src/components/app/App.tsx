import React from 'react';
import ShipsContainer from '../ships/ShipsContainer';
import './App.css';

function App(): JSX.Element {
	return (
		<>
			<ShipsContainer />
			<div className="App">
				Lorem ipsum dolor sit amet, consectetur adipiscing elit. Vestibulum a
				ipsum vel lorem gravida ultricies. Donec ac tempor sem. Pellentesque
				habitant morbi tristique senectus et netus et malesuada fames ac turpis
				egestas. Class aptent taciti sociosqu ad litora torquent per conubia
				nostra, per inceptos himenaeos. Etiam a suscipit eros. Etiam aliquam leo
				non sem auctor sodales. Ut blandit semper mattis. Duis placerat, diam
				ultricies bibendum mattis, justo ex aliquet ligula, ornare posuere ante
				est quis odio. Fusce vestibulum lobortis malesuada. Maecenas imperdiet
				leo vitae dolor sodales, porttitor feugiat lorem fermentum. Nullam
				ornare mi eu pharetra finibus. Curabitur interdum a tellus a lacinia.
				Curabitur mattis dui ante, viverra luctus odio vulputate vitae. Nunc
				mauris est, pretium nec urna nec, dapibus blandit urna. Phasellus
				gravida tempor nisi. Vestibulum non nibh lacus. Nunc accumsan varius
				imperdiet. Donec et cursus tortor, non fermentum urna. Donec finibus
				mattis egestas. In varius mi at nibh tincidunt facilisis. Ut vitae lorem
				dictum, iaculis eros eget, consectetur purus. Donec pretium vitae odio
				et laoreet. Sed vel dolor dolor.
			</div>
		</>
	);
}

export default App;
