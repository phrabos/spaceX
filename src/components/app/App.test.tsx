import React from 'react';
import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import App from './App';
import { rest } from 'msw';
import { setupServer } from 'msw/node';
import rocketData from '../../fixtures/rocketData.json';

const URL = 'https://api.spacexdata.com/v4/rockets';

const server = setupServer(
	rest.get(URL, (req, res, ctx) => {
		return res(ctx.json(rocketData));
	})
);

describe('AnimalCross container', () => {
	beforeAll(() => server.listen());
	afterAll(() => server.close());

	it('renders a list of rockets on the page', async () => {
		render(<App />);

		screen.getByText('Loading...');

		const ul = await screen.findByRole('list', { name: 'rocket-list' });

		expect(ul).not.toBeEmptyDOMElement();
		expect(ul).toMatchSnapshot();
	});
});
