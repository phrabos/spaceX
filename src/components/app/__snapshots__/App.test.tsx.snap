// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AnimalCross container renders a list of rockets on the page 1`] = `
<ul
  aria-label="rocket-list"
>
  <section>
    <li>
      Falcon 1
    </li>
    <img
      alt="Falcon 1"
      src="https://imgur.com/azYafd8.jpg"
      style="width: 300px;"
    />
  </section>
  <section>
    <li>
      Falcon 9
    </li>
    <img
      alt="Falcon 9"
      src="https://farm4.staticflickr.com/3955/32915197674_eee74d81bb_b.jpg"
      style="width: 300px;"
    />
  </section>
  <section>
    <li>
      Falcon Heavy
    </li>
    <img
      alt="Falcon Heavy"
      src="https://farm5.staticflickr.com/4645/38583830575_3f0f7215e6_b.jpg"
      style="width: 300px;"
    />
  </section>
  <section>
    <li>
      Starship
    </li>
    <img
      alt="Starship"
      src="https://live.staticflickr.com/65535/48953946911_e60c5bcc5c_b.jpg"
      style="width: 300px;"
    />
  </section>
</ul>
`;
